Configure.cmake

```diff
@@ -0,0 +1,135 @@
+include(CheckCCompilerFlag)
+include(CheckCSourceCompiles)
+include(CheckTypeSize)
+
+# PLATFORM DETECTION
+
+if(WIN32)
+  set(_TMP_OSTYPE "MSVC")
+elseif(MSYS OR MINGW OR CYGWIN)
+  set(_TMP_OSTYPE "MinGW")
+elseif(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
+  set(_TMP_OSTYPE "Linux")
+elseif(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+  set(_TMP_OSTYPE "Darwin")
+endif()
+
+# Cache OS type into a sleef variable and define it in the configuration header
+set(SLEEF_OSTYPE ${_TMP_OSTYPE} CACHE INTERNAL
+  "The OS 'type' for which SLEEF is configured.")
+
+# Cache machine architecture determined by checking size of a pointer
+if(CMAKE_SIZEOF_VOID_P EQUAL 8)
+  set(SLEEF_ARCH_64BIT ON CACHE INTERNAL
+    "True for 64-bit architecture machines.")
+elseif(CMAKE_SIZEOF_VOID_P EQUAL 4)
+  set(SLEEF_ARCH_32BIT ON CACHE INTERNAL
+    "True for 32-bit architecture machines.")
+endif()
+
+if(CMAKE_SYSTEM_PROCESSOR MATCHES "x86")
+  set(SLEEF_ARCH_X86 ON CACHE INTERNAL "True for x86 architecture.")
+  set(COMPILER_SUPPORTS_SSE2 1)
+endif()
+
+# COMPILER DETECTION
+
+# All variables storing compiler flags should be prefixed with FLAGS_
+
+if(CMAKE_C_COMPILER_ID MATCHES "(GNU|Clang)")
+  set(FLAGS_WALL "-Wall -Wno-unused -Wno-attributes")
+  set(FLAGS_FASTMATH "-ffast-math")
+  set(FLAGS_STRICTMATH "-ffp-contract=off")
+  set(FLAGS_OPENMP "-fopenmp")
+  # Architecture flags for Clang only (TODO: add GNU, ICC, MSVC)
+  set(FLAGS_ENABLE_NEON "-mpfu=neon")
+  set(FLAGS_ENABLE_ADVSIMD "-march=armv8-a+simd")
+  set(FLAGS_ENABLE_SSE2 "-msse2")
+  set(FLAGS_ENABLE_AVX "-mavx")
+  set(FLAGS_ENABLE_AVX2 "-mavx2;-mfma")
+  set(FLAGS_ENABLE_AVX512F "-mavx512f")
+  set(FLAGS_ENABLE_FMA4 "-mfma4")
+elseif(CMAKE_C_COMPILER_ID MATCHES "Intel")
+  # TODO
+elseif(CMAKE_C_COMPILER_ID MATCHES "MSVC")
+  # TODO
+endif()
+
+
+# Warning flags tuning
+# Do we need this for clang; currently get a unknown warning option on Ubuntu
+#if(CMAKE_C_COMPILER_ID MATCHES "Clang")
+  #set(FLAGS_WALL "${FLAGS_WALL} -Wno-shift-negative-value")
+#endif()
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${FLAGS_WALL}")
+
+# Always compile sleef with -ffp-contract and log at configuration time
+if(SLEEF_SHOW_CONFIG)
+  message(STATUS "Using option `${FLAGS_STRICTMATH}` to compile libsleef")
+endif(SLEEF_SHOW_CONFIG)
+
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${FLAGS_STRICTMATH}")
+
+# With -fPIC
+check_c_compiler_flag("-fPIC" WITH_FPIC)
+if (WITH_FPIC)
+  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC")
+endif (WITH_FPIC)
+# TODO: find out why cmake will fail this test with clang on Ubuntu
+# Until then, always use -fPIC
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC")
```

Comment:
Unfortunately, if you need position independent code outside of a shared library, you need to pass in the flag. I wish Cmake was that much magic. The inconvenience is that it fails to detect the fpic on Ubuntu/Clang and so I need to specify it every time, which is a bit annoying. I will remove it from the general CMAKE_C_FLAGS and only add it to the compilation for each individual simd object file.